appkit/src/test/scala/org/ergoplatform/appkit/BabelFeeSpec.scala

```diff
@@ -0,0 +1,168 @@
+      |    val ergPricePerToken: Long = SELF.R5[Long].get
+      |    val tokenId: Coll[Byte] = _tokenId
+      |    val recreatedBabelBoxIndex: Option[Int] = getVar[Int](0)
+      |
+      |    // ===== Perform Babel Fee Swap ===== //
+      |    if (recreatedBabelBoxIndex.isDefined) {
+      |
+      |        // Check conditions for a valid babel fee swap
+      |        val validBabelFeeSwap: Boolean = {
+      |
+      |            // Output babel fee box
+      |            val recreatedBabelBox: Box = OUTPUTS(recreatedBabelBoxIndex.get)
+      |
+      |            // Check that the babel fee box is recreated correctly
+      |            val validBabelFeeBoxRecreation: Boolean =
+      |
+      |                allOf(Coll(
+      |                    (recreatedBabelBox.propositionBytes == SELF.propositionBytes),
+      |                    (recreatedBabelBox.tokens(0)._1 == tokenId),
+      |                    (recreatedBabelBox.R4[SigmaProp].get == babelFeeBoxCreator),
+      |                    (recreatedBabelBox.R5[Long].get == ergPricePerToken),
+      |                    (recreatedBabelBox.R6[Coll[Byte]].get == SELF.id)
+      |                ))
+      |
+      |
+      |
+      |            // Check that the user's token was exchanged correctly
+      |            val validBabelFeeExchange: Boolean = {
+      |
+      |                val nanoErgsDifference: Long = SELF.value - recreatedBabelBox.value
+      |                val babelTokensBefore: Long = if (SELF.tokens.size > 0) SELF.tokens(0)._2 else 0L
+      |                val babelTokensDifference: Long = recreatedBabelBox.tokens(0)._2 - babelTokensBefore
+      |
+      |                allOf(Coll(
+      |                    (babelTokensDifference * ergPricePerToken >= nanoErgsDifference),
+      |                    (nanoErgsDifference >= 0)
+      |                ))
+      |
+      |            }
+      |
+      |            allOf(Coll(
+      |                validBabelFeeBoxRecreation,
+      |                validBabelFeeExchange
+      |            ))
+      |
+      |        }
+      |
+      |        sigmaProp(validBabelFeeSwap)
+      |
+      |    } else {
+      |
+      |        // ===== Perform Babel Fee Box Withdrawl ===== //
+      |        babelFeeBoxCreator
+      |
+      |    }
+      |
+      |}""".stripMargin
+
+  val mockTokenId = "f9e5ce5aa0d95f5d54a7bc89c46730d9662397067250aa18a0039631c0f5b809"
+
+  property("Compile contract for certain token") {
+    val ergoClient = new ColdErgoClient(NetworkType.MAINNET, 0)
+    val contract = ergoClient.execute { ctx: BlockchainContext =>
+      ctx.compileContract(
+        ConstantsBuilder.create()
+          .item("_tokenId", ErgoId.create(mockTokenId).getBytes)
+          .build(),
+        script
+      )
+    }
+
+    val contract2 = new BabelFeeBoxContract().getContractForToken(ErgoId.create(mockTokenId), NetworkType.MAINNET)
+    contract.getErgoTree shouldBe contract2.getErgoTree
```

Comment:
Well, here you are testing the compiler is always producing the same ErgoTree. Not just ErgoTree that logically the same, but exactly the same ErgoTree. So, when the compiler will change a bit and this test will fail after upgrade, what you will do? This will be confusing.

Reply:
Now that the EIP has added the ergotree we can remove that part. It was indeed added as an alarm if something changes to make users aware that different babel fee box addresses will be around.